File: src/array/lc283.go
package array

func moveZeroes(nums []int)  {
  k := 0
  for _, num := range nums {
    if num == 0 {
      continue
    }
    nums[k] = num
    k++
  }
  for k < len(nums) {
    nums[k] = 0
    k++
  }
}

var MoveZeros = moveZeroes

File: src/array/lc283.cc
#include <vector>

using namespace std;

class Solution {
 public:
  void moveZeroes(vector<int>& nums) {
    auto k = 0;
    for (auto i = 0; i < nums.size(); i++) {
      if (nums[i] != 0) {
        nums[k] = nums[i];
        k++;
      }
    }
    for (auto i = k; i < nums.size(); i++) {
      nums[i] = 0;
    }
  }
};
